classes/order/Order.php

```diff
@@ -365,6 +365,7 @@ public function getCartProducts()
         $product_id_list = array();
         $products = $this->getProducts();
         foreach ($products as &$product) {
+            $product['id_product'] = $product['product_id'];
```

Comment:
Please add a `if (empty($product['id_product']))` and a comment explaining why you need to do this Something like
`When product has been deleted from catalog id_product is null, but the data is still available in product_id`

Reply:
Also, is it enough to have just the product id, from what I see in the Order class: https://github.com/PrestaShop/PrestaShop/blob/73feab550c2e40452f249a80eb1f573af16dbd50/classes/order/Order.php#L566

The `product_id ` field comes from the `order_detail` table, but if `id_product` is absent it means the product table doesn't have this row any more, so even if you have the id you won't get the product data

Reply:
Ok seems like all the lost data are actually stored in `order_detail` which is why it works :) But the comment is even more important then

Reply:
> Ok seems like all the lost data are actually stored in `order_detail` which is why it works :) But the comment is even more important then

Yeeah, that's the point. There is no place to find a deleted product expect from a cart that contained it before deletion.

Unfortunately, I can't add commits to this PR. Because of a lack of experience with branching, I messed up things in my fork. So I will close this and create another one where I will add a comment( Your suggestion fits ideally with the purpose IMO)

Thank you very much for the time you gave to my PR